sql/core/src/test/resources/sql-tests/inputs/pipe-operators.sql

```diff
@@ -241,6 +241,85 @@ table t
 table t
 |> extend *;
 
+-- SET operators: positive tests.
+---------------------------------
+
+-- Setting with a constant.
+-- The indicated column is not the last column in the table, and the SET operator will replace it
+-- with the new value in its existing position.
+table t
+|> set x = 1;
+
+-- Setting with an attribute.
+table t
+|> set y = x;
+
+-- Setting with an expression.
+table t
+|> extend 1 as z
+|> set z = x + length(y);
+
+-- Setting two times.
+table t
+|> extend 1 as z
+|> extend 2 as zz
+|> set z = x + length(y), zz = x + 1;
+
+table other
+|> extend 3 as c
+|> set a = b, b = c;
+
+-- Setting two times with a lateral reference.
```

Comment:
how about self reference? e.g. `SET c = c + 1`, shall we allow it?

Reply:
further more, if we allow it, do we need the SET keys uniqueness check? People may want to do `SET c = c + 1, c = c * 2`, and this can be well supported because we use one `Project` per SET key.

Reply:
You're right, both of these are possible now. I removed the duplicate key assignment check and these cases work now. I added a couple test cases to cover it.